core/embed/extmod/modtrezorio/modtrezorio-fatfs.h

```diff
@@ -160,7 +159,7 @@ STATIC mp_obj_t mod_trezorio_FatFSFile___exit__(size_t n_args,
   mp_obj_FatFSFile_t *o = MP_OBJ_TO_PTR(args[0]);
   FRESULT res = f_close(&(o->fp));
   if (res != FR_OK) {
-    mp_raise_OSError(fresult_to_errno_table[res]);
+    mp_raise_OSError(res);
```

Comment:
I don't think that it is correct to stick the FRESULT into OSError as the errno parameter. The python documentation didn't help me much in comprehending whether you can do that:
> errno: A numeric error code from the C variable errno.

I would think that it needs to be one of these: https://docs.python.org/2/library/errno.html, or at the very least that the code should be offset by 256+ so that it does not overlap with the standard ones.

Reply:
I don't think this is relevant to us. upy doesn't even expose the errno constants and there doesn't seem to be any usage of OSError elsewhere. (after all "OSError" is technically a misnomer as there is no "OS" on the device, nor are there errnos)

OTOH maybe using a custom exception class, probably _not_ a subclass of OSError, would be a cleaner solution. I don't think it right to do anything like `fresult_to_errno` or trying to conform to Unixy semantics from a library that doesn't use them in the first place.

Reply:
I think it's fundamentally wrong to mix two different families of error codes. They are overlapping in meaning, aren't they? How is the calling application to know who threw the exception? I think the proper solution is to either:
1. Use a custom exception class (not a subclass of OSError).
2. Offset the FRESULT family of error codes, so that it doesn't overlap with the standard ones.

Reply:
I am surprised MicroPython doesn't expose the error codes, but regardless, that's irrelevant. When you catch an OSError with errno 5, how do you know whether it's MP_EIO or FR_NO_PATH? Am I missing something? I mean even in our own SDCard code, in modtrezorio-sdcard.h, we raise `mp_raise_OSError(MP_EIO)`, so how do you tell the difference? Not to mention all the places where micropython raises OSError.

Reply:
see top-level comment

Reply:
Regarding the usage of the Unix errnos by MicroPython, for example running `git grep MP_EIO` in firmware/vendor/micropython gives me 69 matches. Maybe these are all in modules that we currently don't use (I didn't investigate), but this tells me that there is a real possibility of e.g. errno 5 being raised outside of the fatfs and sd-card code.